workers.tf

```diff
@@ -1,10 +1,80 @@
+
+  lifecycle {
+    create_before_destroy = true
+  }
+
+  block_device_mappings {
+    device_name = "${data.aws_ami.eks_worker.root_device_name}"
+
+    ebs {
+      volume_size           = "${lookup(var.worker_groups[count.index], "root_volume_size", local.workers_group_defaults["root_volume_size"])}"
+      volume_type           = "${lookup(var.worker_groups[count.index], "root_volume_type", local.workers_group_defaults["root_volume_type"])}"
+      iops                  = "${lookup(var.worker_groups[count.index], "root_iops", local.workers_group_defaults["root_iops"])}"
+      delete_on_termination = true
+    }
+  }
+}
+
 resource "aws_autoscaling_group" "workers" {
-  name_prefix           = "${aws_eks_cluster.this.name}-${lookup(var.worker_groups[count.index], "name", count.index)}"
-  desired_capacity      = "${lookup(var.worker_groups[count.index], "asg_desired_capacity", local.workers_group_defaults["asg_desired_capacity"])}"
-  max_size              = "${lookup(var.worker_groups[count.index], "asg_max_size", local.workers_group_defaults["asg_max_size"])}"
-  min_size              = "${lookup(var.worker_groups[count.index], "asg_min_size", local.workers_group_defaults["asg_min_size"])}"
-  target_group_arns     = ["${compact(split(",", coalesce(lookup(var.worker_groups[count.index], "target_group_arns", ""), local.workers_group_defaults["target_group_arns"])))}"]
-  launch_configuration  = "${element(aws_launch_configuration.workers.*.id, count.index)}"
+  name_prefix       = "${aws_eks_cluster.this.name}-${lookup(var.worker_groups[count.index], "name", count.index)}"
+  desired_capacity  = "${lookup(var.worker_groups[count.index], "asg_desired_capacity", local.workers_group_defaults["asg_desired_capacity"])}"
+  max_size          = "${lookup(var.worker_groups[count.index], "asg_max_size", local.workers_group_defaults["asg_max_size"])}"
+  min_size          = "${lookup(var.worker_groups[count.index], "asg_min_size", local.workers_group_defaults["asg_min_size"])}"
+  target_group_arns = ["${compact(split(",", coalesce(lookup(var.worker_groups[count.index], "target_group_arns", ""), local.workers_group_defaults["target_group_arns"])))}"]
+
+  mixed_instances_policy {
+    instances_distribution {
+      on_demand_allocation_strategy            = "${var.on_demand_allocation_strategy}"
+      on_demand_base_capacity                  = "${var.on_demand_base_capacity}"
+      on_demand_percentage_above_base_capacity = "${var.on_demand_percentage_above_base_capacity}"
+      spot_allocation_strategy                 = "${var.spot_allocation_strategy}"
+      spot_instance_pools                      = "${var.spot_instance_pools}"
+      spot_max_price                           = "${var.spot_max_price}"
+    }
+
+    launch_template {
+      launch_template_specification {
+        launch_template_id = "${element(aws_launch_template.workers.*.id, count.index)}"
+        version            = "$Latest"
+      }
+
+      override {
```

Comment:
It seems like the AWS provider devs are also waiting on Terraform 0.14 to fix everything as well...

It's currently not possible to make this overrides dynamically generated and you're required to have 2 or more of these overrides.